### PR TITLE
Added some refactories to improve code readability

### PR DESCRIPTION
**Pull request by lockland**

### CODE REVIEW

**Comment by fabpot** [comment]
I'm against those changes. We do not accept changes that don't fix bug and that can potentially make merging older branches more difficult and can also potentially introduce subtle bugs.

File: src/Symfony/Component/Console/Application.php [hunk 1]
```diff
@@ -148,13 +148,8 @@ public function run(InputInterface $input = null, OutputInterface $output = null
...
 
-            $exitCode = $e->getCode();
-            if (is_numeric($exitCode)) {
-                $exitCode = (int) $exitCode;
-                if (0 === $exitCode) {
-                    $exitCode = 1;
-                }
-            } else {
+            $exitCode = (int) $e->getCode();
+            if (! is_numeric($e->getCode()) || 0 === $exitCode) {
```
Comment (Simperfit) [inline src/Symfony/Component/Console/Application.php]: Are you sure that this will cause no bug, i'm not.

**Comment by Simperfit** [comment]
I'm -1 on this too, please do some refactoring when addings new feature.

**Comment by ostrolucky** [comment]
I think refactorings are allowed if target is lowest maintained branch, changes are easy to merge into all maintained branches and patch doesn't noticeably changes current behaviour. None of which is true for this PR. So I'm with others here -1.

**Comment by chalasr** [comment]
Same for me, refactoring should just be avoided if no strong technical benefit.

File: src/Symfony/Component/Console/Application.php (hunk 1 above)
Comment (lockland) [reply src/Symfony/Component/Console/Application.php]: Well, if the test isn't trusted, I am not sure, but I think it was, at least this they must be

**Pull request closed without merging**